packages/assets-controllers/src/TokensController.ts

```diff
@@ -704,10 +746,16 @@ export class TokensController extends BaseController<
             image,
             suggestedAssetMeta?.interactingAddress || selectedAddress,
           );
-          suggestedAssetMeta.status = SuggestedAssetStatus.accepted;
+
+          this._acceptApproval(suggestedAssetID);
+
+          const acceptedSuggestedAssetMeta = {
```

Comment:
I had to remove the line that updates status directly (`suggestedAssetMeta.status = SuggestedAssetStatus.accepted;`) and replace it by a new object using the spread operator, as otherwise that line throws a read error when using this controller in mobile codebase with react-native.

```
Attempted to assign to readonly property
```